warp10/src/main/java/io/warp10/standalone/Warp.java

```diff
@@ -135,16 +136,48 @@ public static void main(String[] args) throws Exception {
 
     Properties properties = getProperties();
 
+    backend = properties.getProperty(Configuration.BACKEND);
+
+    boolean useLevelDB = false;
+    boolean useFDB = false;
+    boolean nullbackend = false;
+    boolean plasmabackend = false;
+    boolean inmemory = false;
+
     boolean analyticsEngineOnly = "true".equals(properties.getProperty(Configuration.ANALYTICS_ENGINE_ONLY));
-    boolean nullbackend = "true".equals(properties.getProperty(NULL)) || analyticsEngineOnly;
 
-    boolean plasmabackend = "true".equals(properties.getProperty(Configuration.PURE_PLASMA));
+    if (analyticsEngineOnly) {
+      nullbackend = true;
+      if (null != backend && !Constants.BACKEND_NULL.equals(backend)) {
+        LOG.warn("Backend specification '" + backend + "' ignored since '" + Configuration.ANALYTICS_ENGINE_ONLY + " is true.");
+      }
+    } else {
+      switch(backend) {
+        case Constants.BACKEND_LEVELDB:
+          useLevelDB = true;
+          break;
+        case Constants.BACKEND_FDB:
+          useFDB = true;
+          break;
+        case Constants.BACKEND_NULL:
+          nullbackend = true;
+          break;
+        case Constants.BACKEND_PLASMA:
+          plasmabackend = true;
+          break;
+        case Constants.BACKEND_MEMORY:
+          inmemory = true;
+          break;
+        default:
+          throw new RuntimeException("Missing '" + Configuration.BACKEND + "' specification.");
+      }
+    }
```

Comment:
switch cannot accept null.
So when backend is undefined (the default, for people who will migrate to 3.x from 2.x standalone), it must be managed before, and should default to leveldb.